zingolib/src/lightclient/send.rs

```diff
@@ -81,38 +111,133 @@ pub mod send_with_proposal {
+                        transactions_to_record.push(Transaction::read(
+                            &raw_tx[..],
+                            zcash_primitives::consensus::BranchId::for_height(
+                                &self.wallet.transaction_context.config.chain,
+                                current_height,
+                            ),
+                        )?);
+                    }
+                }
+            }
+            drop(tx_map);
+            for transaction in transactions_to_record {
+                self.wallet
+                    .transaction_context
+                    .scan_full_tx(
+                        &transaction,
+                        zingo_status::confirmation_status::ConfirmationStatus::Transmitted(
+                            current_height,
+                        ),
+                        Some(now() as u32),
+                        crate::wallet::utils::get_price(
+                            now(),
+                            &self.wallet.price.read().await.clone(),
+                        ),
+                    )
+                    .await;
+            }
+            Ok(())
+        }
+
+        /// Calculates, signs and broadcasts transactions from a proposal.
+        async fn broadcast_created_transactions(
+            &self,
+        ) -> Result<NonEmpty<TxId>, BroadcastCachedTransactionsError> {
+            let mut tx_map = self
+                .wallet
+                .transaction_context
+                .transaction_metadata_set
+                .write()
+                .await;
+            match tx_map.spending_data_mut() {
+                None => Err(BroadcastCachedTransactionsError::Cache(
+                    TransactionCacheError::NoSpendCapability,
+                )),
+                Some(ref mut spending_data) => {
+                    let mut txids = vec![];
+                    for (txid, raw_tx) in spending_data.cached_raw_transactions() {
+                        match crate::grpc_connector::send_transaction(
+                            self.get_server_uri(),
+                            raw_tx.clone().into_boxed_slice(),
+                        )
+                        .await
+                        {
+                            Ok(serverz_txid_string) => {
+                                txids.push(crate::utils::txid::compare_txid_to_string(
+                                    *txid,
+                                    serverz_txid_string,
+                                    self.wallet.transaction_context.config.accept_server_txids,
+                                ))
+                            }
+                            Err(server_err) => {
+                                return Err(BroadcastCachedTransactionsError::Broadcast(server_err))
+                            }
+                        };
+                    }
+
+                    let non_empty_serverz_txids = NonEmpty::from_vec(txids).ok_or(
```

Comment:
the type describes its non-emptiness already

Reply:
fixed